lib/CryptoNoteCore/Miner.cpp

```diff
@@ -249,8 +249,11 @@ bool miner::start(const AccountPublicAddress &adr, size_t threads_count)
     std::lock_guard<std::mutex> lk(m_threads_lock);
 
     if(!m_threads.empty()) {
-        logger(ERROR) << "Unable to start miner because there are active mining threads";
-        return false;
+        m_stop = true;
+        for (auto &th : m_threads) {
+            th.join();
+        }
+        m_threads.clear();
```

Comment:
stopping and clearing the reserved threads is way better than return false

Reply:
This stopping is required to support `generate-blocks`functionality. We can't call usual `stop` for mining after the given blocks number is generated due to the possible deadlock,  we only can send the stop signal. The most comfortable to clear threads after that is to clear them during `start_mining` call.